lxd/storage/drivers/driver_ceph_volumes.go

```diff
@@ -364,21 +364,27 @@ func (d *ceph) CreateVolumeFromCopy(vol Volume, srcVol Volume, copySnapshots boo
 
 	// For VMs, also copy the filesystem volume.
 	if vol.IsVMBlock() {
-		srcFSVol := srcVol.NewVMBlockFilesystemVolume()
-		fsVol := vol.NewVMBlockFilesystemVolume()
-		err := d.CreateVolumeFromCopy(fsVol, srcFSVol, copySnapshots, false, op)
+		srcFSVol := NewVolumeCopy(srcVol.NewVMBlockFilesystemVolume())
+		fsVol := NewVolumeCopy(vol.NewVMBlockFilesystemVolume())
+		err := d.CreateVolumeFromCopy(fsVol, srcFSVol, false, op)
 		if err != nil {
 			return err
 		}
 
 		// Delete on revert.
-		revert.Add(func() { _ = d.DeleteVolume(fsVol, op) })
+		revert.Add(func() { _ = d.DeleteVolume(fsVol.Volume, op) })
+	}
+
+	// Check if there is at least one snapshot that requires copying.
+	copySnapshots := false
```

Comment:
copySnapshots is only used in 1 place AFAIK, so think we can simplify the if statement that uses it below to just check len(vol.Snapshots) > 0 directly.

Reply:
I have also removed the variable now from the other drivers as it can be made simpler everywhere just by using the `len(vol.Snapshots)` comparison once.